File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) 
    {

        if(s.length()==0) return 0;

        Set<Character> set=new HashSet<>();
        int max=Integer.MIN_VALUE;

        int i=0;
        int j=0;

        while(j<s.length())
        {
            while(set.contains(s.charAt(j)))
            {
                set.remove(s.charAt(i));
                i++;
            }

            set.add(s.charAt(j++));
            max=Math.max(max,j-i);
        }

        return max;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.kt
class Solution {
    fun lengthOfLongestSubstring(s: String): Int 
    {
        val set=HashSet<Char>()

        var i=0
        var j=0

        var ans=0

        while(j<s.length)
        {
            while(set.contains(s[j]))
            {
                set.remove(s[i])
                i++
            }

            set.add(s[j])
            j++

            ans=max(ans,(j-i))
        }


        return ans
    }
}